paymentsheet/src/main/java/com/stripe/android/link/LinkControllerViewModel.kt

```diff
@@ -99,62 +105,30 @@ internal class LinkControllerViewModel @Inject constructor(
         launcher: ActivityResultLauncher<LinkActivityContract.Args>,
         email: String?
     ) {
-        if (presentJob?.isActive == true) {
-            logger.debug("$tag: already presenting")
-            return
-        }
-        presentJob = viewModelScope.launch {
-            logger.debug("$tag: presenting payment methods")
-
-            withConfiguration(
-                email = email,
-                onError = { error ->
-                    _presentPaymentMethodsResultFlow.emit(
-                        LinkController.PresentPaymentMethodsResult.Failed(error)
-                    )
-                },
-                onSuccess = { configuration ->
-                    val state = _state.value
-                    val linkAccountInfo = getLinkAccountInfo(email, state)
-
-                    // If the account changed, clear account-related state.
-                    val selectedPaymentMethod = state.selectedPaymentMethod
-                        .takeIf { linkAccountInfo.account != null }
-                    val createdPaymentMethod = state.createdPaymentMethod
-                        .takeIf { linkAccountInfo.account != null }
-
-                    val launchMode = LinkLaunchMode.PaymentMethodSelection(selectedPaymentMethod?.details)
-
-                    _state.update {
-                        it.copy(
-                            presentedForEmail = email,
-                            selectedPaymentMethod = selectedPaymentMethod,
-                            createdPaymentMethod = createdPaymentMethod,
-                            currentLaunchMode = launchMode,
-                        )
-                    }
-
-                    launcher.launch(
-                        LinkActivityContract.Args(
-                            configuration = configuration,
-                            startWithVerificationDialog = true,
-                            linkAccountInfo = linkAccountInfo,
-                            launchMode = launchMode,
-                        )
-                    )
-                }
-            )
-        }
+        present(
```

Comment:
👀 Extracted a `present()` method that centralized state cleanup on account/email changes.